airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/base/IntegrationRunnerTest.java

```diff
@@ -275,4 +293,97 @@ void testDestinationConsumerLifecycleFailure() throws Exception {
     }
   }
 
+  @Test
+  void testInterruptOrphanThreadFailure() {
+    System.setIn(new ByteArrayInputStream("{}\n{}".getBytes()));
+    final String testName = Thread.currentThread().getName();
+
+    try (final MultiThreadTestConsumer airbyteMessageConsumerMock = new MultiThreadTestConsumer(false)) {
+      assertThrows(IOException.class, () -> IntegrationRunner.consumeWriteStream(airbyteMessageConsumerMock, true,
+          3, TimeUnit.SECONDS,
+          10, TimeUnit.SECONDS));
+      try {
+        TimeUnit.SECONDS.sleep(10);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+      final List<Thread> runningThreads = ThreadUtils.getAllThreads().stream()
+          .filter(runningThread -> !runningThread.isDaemon() && !runningThread.getName().equals(testName))
+          .collect(Collectors.toList());
+      // all threads should be interrupted
+      assertEquals(List.of(), runningThreads);
+      assertTrue(airbyteMessageConsumerMock.hasCaughtExceptions());
+
+    }
+  }
+
+  @Test
+  void testNoInterruptOrphanThreadFailure() {
```

Comment:
Yes, I tried to test the System.exit call in this test because we can have connector code or when we use certain libraries that catch and swallow interrupt exceptions too... 

It was based on catching `System.exit()` calls using SecurityManager class in java: https://stackoverflow.com/questions/5549720/how-to-prevent-calls-to-system-exit-from-terminating-the-jvm

But it turns out, with java 17, using this solution triggers a warning:
`WARNING: A terminally deprecated method in java.lang.System has been called`

resulting in a compilation error with gradle so I had to revert that part out. It doesn't seem like there are alternatives method to reproduce the same feature for the moment.

Reply:
got it. does that mean this test case doesn't do much then?

Reply:
Good point, I refactored it so I can test it more easily

thanks

Reply:
synced offline - made sure we were able to change the method signature so a mock exit is called.